### PR TITLE
Delete cards from code mode

### PR DESCRIPTION
**Pull request by habdelra**
This PR introduces a delete button from code mode.

@tintinthong note that as part of this i removed all the speculative code around how we pass actions into the definition container and hiding/showing buttons (since there is only one actual button with real functionality). Add that implementation when we have real functionality behind it--otherwise it's hard to refactor and just deal with in general when there is nothing backing it functionality or from a test perspective.

Here is a screen capture when there are not recent files and we perform a delete. Note that we show the empty code mode, and when you go back to interact mode the stack item has been removed.

delete-code-mode

Here is a screen capture when there is a recent file and we perform a delete. in this case we show the previously opened file from code mode after the delete, and when you go back to interact mode the stack item has been removed.

delete-code-mode2

### CODE REVIEW

**Comment by tintinthong** [comment]
> @tintinthong note that as part of this i removed all the speculative code around how we pass actions into the definition container and hiding/showing buttons (since there is only one actual button with real functionality). Add that implementation when we have real functionality behind it--otherwise it's hard to refactor and just deal with in general when there is nothing backing it functionality or from a test perspective.

I made some upstream changes and resolved them in this pr

**Review by tintinthong — APPROVED** [review]
LGTM

File: packages/host/tests/acceptance/operator-mode-test.ts [hunk 1]
```diff
@@ -1321,6 +1329,195 @@ module('Acceptance | operator mode tests', function (hooks) {
...
+      realm,
+      adapter,
+      expectedEvents,
+      async () => {
+        await click('[data-test-confirm-delete-button]');
+      },
+    );
+    await waitUntil(() => find('[data-test-editor]'));
+    // TODO make an assertion around the URL bar when the bug for
+    // updating the URL bar correctly is fixed
```
Comment (backspace) [inline packages/host/tests/acceptance/operator-mode-test.ts:1491]: is this tracked anywhere else?

**Review by backspace — APPROVED** [review]
👍🏻

maybe this addresses issue 5910?

File: packages/host/tests/acceptance/operator-mode-test.ts (hunk 1 above)
Comment (habdelra) [reply packages/host/tests/acceptance/operator-mode-test.ts:1491]: I think this is a PR that @tintinthong was working on. maybe @tintinthong  has better thoughts here how to track this?

**Comment by habdelra** [comment]
> 👍🏻
> 
> maybe this addresses issue 5910?

it should, recent files now consumes the whole URL

File: packages/host/tests/acceptance/operator-mode-test.ts (hunk 1 above)
Comment (habdelra) [reply packages/host/tests/acceptance/operator-mode-test.ts:1491]: (worst case scenario we can just write a ticket)

Comment (tintinthong) [reply packages/host/tests/acceptance/operator-mode-test.ts:1491]: I'll resolve this. I think its just a matter of adding assertion

**Pull request merged**